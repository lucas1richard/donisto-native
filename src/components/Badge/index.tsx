import * as React from 'react';
import { View } from 'react-native';
import Txt from 'components/Txt';
import badgeStyles from './styles';

interface IBadgeProps {
  label: string;
}

class Badge extends React.PureComponent<IBadgeProps> {
  render() {
    const { label } = this.props;
    return (
      <View style={badgeStyles.wrapper}>
        <Txt color="light">
          {label}
        </Txt>
      </View>
    );
  }
}

export default Badge;