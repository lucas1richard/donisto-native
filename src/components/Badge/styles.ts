import { StyleSheet } from 'react-native';

const badgeStyles = StyleSheet.create({
  wrapper: {
    borderRadius: 999,
    paddingHorizontal: 10,
    paddingVertical: 3,
    backgroundColor: 'rgba(0,0,0,0.6)',
  },
});

export default badgeStyles;