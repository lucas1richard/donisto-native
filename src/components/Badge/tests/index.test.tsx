import * as React from 'react';
import { shallow } from 'enzyme';
import Badge, { IBadgeProps } from '..';

describe('<Badge />', () => {
  const renderComponent = (props: IBadgeProps) => shallow(<Badge {...props} />);
  it('renders', () => {
    expect(renderComponent({
      label: 'mockLabel'
    })).toMatchSnapshot();
  });
});