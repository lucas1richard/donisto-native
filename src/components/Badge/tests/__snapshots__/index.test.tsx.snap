// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Badge
    label="mockLabel"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Txt
        color="light"
      >
        mockLabel
      </Txt>,
      "style": Object {
        "backgroundColor": "rgba(0,0,0,0.6)",
        "borderRadius": 999,
        "paddingHorizontal": 10,
        "paddingVertical": 3,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "mockLabel",
        "color": "light",
      },
      "ref": null,
      "rendered": "mockLabel",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Txt
          color="light"
        >
          mockLabel
        </Txt>,
        "style": Object {
          "backgroundColor": "rgba(0,0,0,0.6)",
          "borderRadius": 999,
          "paddingHorizontal": 10,
          "paddingVertical": 3,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "mockLabel",
          "color": "light",
        },
        "ref": null,
        "rendered": "mockLabel",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
